packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -124,7 +127,11 @@ type SchemaDerivedData = {
 type ServerState =
   | { phase: 'initialized with schema'; schemaDerivedData: SchemaDerivedData }
   | { phase: 'initialized with gateway'; gateway: GraphQLService }
-  | { phase: 'starting'; barrier: Resolvable<void> }
+  | {
+      phase: 'starting';
+      barrier: Resolvable<void>;
+      schemaDerivedData?: SchemaDerivedData;
```

Comment:
To me, the difference between the "starting" and "invoking serverWillStart" states is that the latter has the (initial) schema set up and derived, and the former doesn't. I'm not sure what the point is of having them as separate states otherwise.  I guess you could merge them.  But really I think it shouldn't be rocket science for us to get it working with distinct states that represent distinct moments in moving forward?

Reply:
I ended up eliding out the `starting` since it was never distinguished from `invoking serverWillStart` in the updated PR.